src/Analyzers/CSharp/Analyzers/MakeStructMemberReadOnly/CSharpMakeStructMemberReadOnlyAnalyzer.cs

```diff
@@ -68,18 +97,9 @@ protected override void InitializeWorker(AnalysisContext context)
     {
         var cancellationToken = context.CancellationToken;
 
-        // if it's not a method, or it's already readonly, nothing to do.
-        if (context.OwningSymbol is not IMethodSymbol
-            {
-                MethodKind: MethodKind.Ordinary or MethodKind.ExplicitInterfaceImplementation or MethodKind.PropertyGet or MethodKind.PropertySet,
-                IsReadOnly: false,
-                IsStatic: false,
-                IsImplicitlyDeclared: false,
-                DeclaringSyntaxReferences: [var methodReference, ..],
-            } owningMethod)
-        {
+        var location = GetDiagnosticLocation(context.OwningSymbol, cancellationToken, out var additionalLocation);
+        if (location == null || !context.ShouldAnalyzeSpan(location.SourceSpan))
```

Comment:
not understanding the `location/additionalLocation` split.

Reply:
Retaining the original semantics with just refactored code. We only check whether or not the primary location is within the analysis span.